src/components/Checkbox/index.test.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import { render, screen, within } from '@testing-library/react';
+
+import Checkbox, { checkboxDataTestIds } from '.';
+
+describe('Checkbox', () => {
+  describe('given the checkbox is checked', () => {
+    it('renders a checkbox component as checked', () => {
+      render(<Checkbox isValueChecked />);
+
+      const checkBox = screen.getByTestId(checkboxDataTestIds.base);
+
+      expect(checkBox).toBeVisible();
+      expect(within(checkBox).getByRole('checkbox')).toBeChecked();
+    });
+  });
+
+  describe('given the checkbox is unchecked', () => {
+    it('renders a checkbox component as unchecked', () => {
+      render(<Checkbox isValueChecked={false} />);
+
+      const checkBox = screen.getByTestId(checkboxDataTestIds.base);
+
+      expect(within(checkBox).getByRole('checkbox')).not.toBeChecked();
```

Comment:
Should we check if the checkbox is rendered as well?
```suggestion
      expect(checkBox).toBeVisible();
      expect(within(checkBox).getByRole('checkbox')).not.toBeChecked();
```

Reply:
@hoangmirs updated in [9cf9cc7](https://github.com/manh-t/react-survey/pull/79/commits/9cf9cc7e64b19e65e275327d6b0981dc4379edc3)